x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/cache/shared/NodesCachesStatsIntegTests.java

```diff
@@ -0,0 +1,161 @@
+            final long totalFsSize = client().admin()
+                .cluster()
+                .prepareNodesStats(nodeId)
+                .clear()
+                .setFs(true)
+                .get()
+                .getNodesMap()
+                .get(nodeId)
+                .getFs()
+                .getTotal()
+                .getTotal()
+                .getBytes();
+
+            final long cacheSize = FrozenCacheService.calculateCacheSize(clusterService.getSettings(), totalFsSize);
+            assertThat(nodeCachesStats.getSize(), equalTo(cacheSize));
+
+            final long regionSize = FrozenCacheService.SNAPSHOT_CACHE_REGION_SIZE_SETTING.get(clusterService.getSettings()).getBytes();
+            assertThat(nodeCachesStats.getRegionSize(), equalTo(regionSize));
+
+            assertThat(nodeCachesStats.getNumRegions(), equalTo(Math.toIntExact(cacheSize / regionSize)));
+            assertThat(nodeCachesStats.getWrites(), equalTo(0L));
+            assertThat(nodeCachesStats.getBytesWritten(), equalTo(0L));
+            assertThat(nodeCachesStats.getReads(), equalTo(0L));
+            assertThat(nodeCachesStats.getBytesRead(), equalTo(0L));
+            assertThat(nodeCachesStats.getEvictions(), equalTo(0L));
+        }
+
+        final ClearSearchableSnapshotsCacheResponse clearCacheResponse = client().execute(
+            ClearSearchableSnapshotsCacheAction.INSTANCE,
+            new ClearSearchableSnapshotsCacheRequest(mountedIndex)
+        ).actionGet();
+        assertThat(clearCacheResponse.getSuccessfulShards(), greaterThan(0));
+        assertThat(clearCacheResponse.getFailedShards(), equalTo(0));
+
+        for (int i = 0; i < 20; i++) {
+            client().prepareSearch(mountedIndex)
+                .setQuery(
+                    randomBoolean()
+                        ? QueryBuilders.rangeQuery("id").gte(randomIntBetween(0, 1000))
+                        : QueryBuilders.termQuery("test", "value" + randomIntBetween(0, 1000))
+                )
+                .setSize(randomIntBetween(0, 1000))
+                .get();
+        }
+
+        assertExecutorIsIdle(SearchableSnapshotsConstants.CACHE_FETCH_ASYNC_THREAD_POOL_NAME);
+
+        final Set<String> dataNodesWithFrozenShards = client().admin()
+            .cluster()
+            .prepareState()
+            .get()
+            .getState()
+            .routingTable()
+            .index(mountedIndex)
+            .shardsWithState(ShardRoutingState.STARTED)
+            .stream()
+            .filter(ShardRouting::assignedToNode)
+            .map(ShardRouting::currentNodeId)
+            .collect(Collectors.toSet());
+
+        final NodesCachesStatsResponse response = client().execute(
+            TransportSearchableSnapshotsNodeCachesStatsAction.TYPE,
+            new NodesRequest(dataNodesWithFrozenShards.toArray(String[]::new))
+        ).actionGet();
+        assertThat(response.getNodes(), hasSize(dataNodesWithFrozenShards.size()));
```

Comment:
nit: maybe we can assert that the returned list of nodes is the same as the requested instead of comparing their size.

Reply:
Sure, I pushed 2611bd94099220e39ab30f253e2ee5e4d0fee8ce